watertap/costing/units/gac.py

```diff
@@ -178,6 +123,50 @@ def cost_gac(blk, contactor_type=ContactorType.pressure):
         contactor_type: ContactorType Enum indicating whether to cost based on steel pressure vessels or concrete,
             default = ContactorType.pressure
     """
+
+    # ---------------------------------------------------------------------
+    # with ContactorType not assigned in build_gac_cost_param_block and blk.costing_package.gac variables
+    # fixed when register_costing_parameter_block, refix parameters based on contactor type here
+
+    # costing data parameters based on contactor type
+    adsorbent_unit_cost_coeff_data = {0: 4.58342, 1: -1.25311e-5}
+    if contactor_type == ContactorType.pressure:
+        contactor_cost_coeff_data = {0: 10010.9, 1: 2204.95, 2: -15.9378, 3: 0.110592}
+        other_cost_param_data = {0: 16660.7, 1: 0.552207}
+        energy_consumption_coeff_data = {0: 8.09926e-4, 1: 8.70577e-4, 2: 0}
+    elif contactor_type == ContactorType.gravity:
+        contactor_cost_coeff_data = {0: 75131.3, 1: 735.550, 2: -1.01827, 3: 0.000000}
+        other_cost_param_data = {0: 38846.9, 1: 0.490571}
+        energy_consumption_coeff_data = {0: 0.123782, 1: 0.132403, 2: -1.41512e-5}
+    else:
+        raise ConfigurationError(
+            f"{blk.unit_model.name} received invalid argument for contactor_type:"
+            f" {contactor_type}. Argument must be a member of the ContactorType Enum."
+        )
+
+    # iterable lists with matching indices
+    cost_param_data_list = [
+        adsorbent_unit_cost_coeff_data,
+        contactor_cost_coeff_data,
+        other_cost_param_data,
+        energy_consumption_coeff_data,
+    ]
+    cost_param_var_list = [
+        blk.costing_package.gac.adsorbent_unit_cost_coeff,
+        blk.costing_package.gac.contactor_cost_coeff,
+        blk.costing_package.gac.other_cost_param,
+        blk.costing_package.gac.energy_consumption_coeff,
+    ]
+
+    # refix variables to appropriate costing parameters
+    for index in range(len(cost_param_var_list)):
+        param_data = cost_param_data_list[index]
+        param_var = cost_param_var_list[index]
+        for param_index in param_var.keys():
+            param_var[param_index].fix(param_data[param_index])
```

Comment:
It's a small thing, but you could use `ComponentMap` here to map the var to the coeff data.

Alternatively, you could create a similar `tuple`, e.g., `[( blk.costing_package.gac.adsorbent_unit_cost_coeff, adsorbent_unit_cost_coeff_data), ...]`.

Or, we could use `zip` to iterate over the list simultaneously. This is my least favorite option. Ideally the var would be close to the coeff_data in the code.

Reply:
I thought the same thing originally with the tuple, but figured it wouldn't actually consolidate anything because black would end up formatting it in a similar way. I will look into the `ComponentMap` briefly though.

Reply:
The point with the `tuples` is that it's easier to catch mistakes because the two pieces of information are together in the code. Same with `ComponentMap`.